homeassistant/components/whirlpool/climate.py

```diff
@@ -116,7 +116,7 @@ def __init__(self, hass, said, name, backend_selector: BackendSelector, auth: Au
         self._said = said
         self._attr_device_info = DeviceInfo(
             identifiers={(DOMAIN, self._said)},
-            name=name,
+            name=name if name is not None else said,
```

Comment:
This means the device name & entity name are the same (and thus in conflict with the `has_entity_name` set).

This integration has a unique ID, so maybe I'm confused on the backwards compatibility issue you are trying to address?

Reply:
Maybe I misunderstand too.  
My understanding: FriendlyName comes from device name + entity name when _attr_has_entity_name is True.  Previously, the name was created by just _attr__name.  By setting the device name equal to the old name, and not setting _attr_name, we get the same Friendly Name as before adding DeviceInfo.

The other way around would be to set the device name = SAID (unique per device) and _attr_name = name (or said if none) Name would then be SAID + name or SAID + SAID (a bit odd name)

Additionally, if name = None, we need something in one of the names (don't we?)